File: generated_tests/pascal/tests/windows_tests.pas
uses SplashKit, TestFramework, ../Helpers;
type
    TTestWindows = class(TTestCase)
    protected
        procedure Setup; override;
    end;
    procedure TestClearWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 1', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    color := ColorBlack();
    ClearWindow(testWindow, color);
    RefreshWindow(testWindow);
    pixel := GetPixel(PointAt(0.0, 0.0));
    AssertEquals(color, pixel);
end;
procedure TestCloseAllWindowsIntegration;
begin
    OpenWindow('Test Window 2', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    OpenWindow('Test Window 3', 800, 600);
    CloseAllWindows();
    AssertFalse(HasWindow('Test Window 3'));
    AssertFalse(HasWindow('Test Window 3'));
end;
procedure TestCloseCurrentWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 4', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    CloseCurrentWindow();
    AssertFalse(HasWindow('Test Window 4'));
    AssertNull(CurrentWindow());
end;
procedure TestCloseWindowNamedIntegration;
begin
    OpenWindow('Test Window 5', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertTrue(HasWindow('Test Window 5'));
    CloseWindow('Test Window 5');
    AssertFalse(HasWindow('Test Window 5'));
end;
procedure TestCloseWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 6', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    windowName := WindowCaption(testWindow);
    CloseWindow(testWindow);
    AssertFalse(HasWindow(windowName));
end;
procedure TestCurrentWindowIntegration;
begin
    testWindow1 := OpenWindow('Test Window 7', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    testWindow2 := OpenWindow('Test Window 8', 800, 600);
    SetCurrentWindow(testWindow2);
    AssertEquals(testWindow2, CurrentWindow());
    AssertTrue(IsCurrentWindow(testWindow2));
    AssertFalse(IsCurrentWindow(testWindow1));
end;
procedure TestCurrentWindowHasBorderIntegration;
begin
    testWindow := OpenWindow('Test Window 9', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    AssertTrue(CurrentWindowHasBorder());
    CurrentWindowToggleBorder();
    AssertFalse(CurrentWindowHasBorder());
end;
procedure TestCurrentWindowHeightIntegration;
begin
    testWindow := OpenWindow('Test Window 10', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    AssertEquals(600, CurrentWindowHeight());
    ResizeCurrentWindow(800, 400);
    AssertEquals(400, CurrentWindowHeight());
end;
procedure TestCurrentWindowIsFullscreenIntegration;
begin
    testWindow := OpenWindow('Test Window 11', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    AssertFalse(CurrentWindowIsFullscreen());
    CurrentWindowToggleFullscreen();
    AssertTrue(CurrentWindowIsFullscreen());
end;
procedure TestCurrentWindowPositionIntegration;
begin
    testWindow := OpenWindow('Test Window 12', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    MoveCurrentWindowTo(100, 200);
    position := CurrentWindowPosition();
    AssertEquals(68.0, position.x);
    AssertEquals(168.0, position.y);
end;
procedure TestCurrentWindowToggleBorderIntegration;
begin
    testWindow := OpenWindow('Test Window 13', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    initialBorder := CurrentWindowHasBorder();
    CurrentWindowToggleBorder();
    AssertNotEquals(initialBorder, CurrentWindowHasBorder());
end;
procedure TestCurrentWindowToggleFullscreenIntegration;
begin
    testWindow := OpenWindow('Test Window 14', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    initialFullscreen := CurrentWindowIsFullscreen();
    CurrentWindowToggleFullscreen();
    AssertNotEquals(initialFullscreen, CurrentWindowIsFullscreen());
end;
procedure TestCurrentWindowWidthIntegration;
begin
    testWindow := OpenWindow('Test Window 15', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    AssertEquals(800, CurrentWindowWidth());
    ResizeCurrentWindow(400, 600);
    AssertEquals(400, CurrentWindowWidth());
end;
procedure TestCurrentWindowXIntegration;
begin
    testWindow := OpenWindow('Test Window 16', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    MoveCurrentWindowTo(100, 200);
    AssertEquals(68, CurrentWindowX());
    MoveCurrentWindowTo(300, 200);
    AssertEquals(268, CurrentWindowX());
end;
procedure TestCurrentWindowYIntegration;
begin
    testWindow := OpenWindow('Test Window 17', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    MoveCurrentWindowTo(0, 100);
    AssertEquals(68, CurrentWindowY());
end;
procedure TestHasWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 18', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertTrue(HasWindow('Test Window 18'));
    AssertTrue(HasWindow(WindowCaption(testWindow)));
    CloseWindow(testWindow);
    AssertFalse(HasWindow('Test Window 18'));
end;
procedure TestIsCurrentWindowIntegration;
begin
    testWindow1 := OpenWindow('Test Window 19', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    testWindow2 := OpenWindow('Test Window 20', 800, 600);
    SetCurrentWindow(testWindow1);
    AssertTrue(IsCurrentWindow(testWindow1));
    AssertFalse(IsCurrentWindow(testWindow2));
end;
procedure TestMoveCurrentWindowToIntegration;
begin
    testWindow := OpenWindow('Test Window 21', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    MoveCurrentWindowTo(100, 200);
    AssertEquals(68, CurrentWindowX());
    AssertEquals(168, CurrentWindowY());
end;
procedure TestMoveWindowToNamedIntegration;
begin
    OpenWindow('Test Window 22', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo('Test Window 22', 150, 250);
    AssertEquals(118, WindowX('Test Window 22'));
    AssertEquals(218, WindowY('Test Window 22'));
end;
procedure TestMoveWindowToIntegration;
begin
    testWindow := OpenWindow('Test Window 23', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo(testWindow, 200, 300);
    AssertEquals(168, WindowX(testWindow));
    AssertEquals(268, WindowY(testWindow));
end;
procedure TestOpenWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 24', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertNotNull(testWindow);
    AssertEquals('Test Window 24', WindowCaption(testWindow));
    AssertEquals(800, WindowWidth(testWindow));
    AssertEquals(600, WindowHeight(testWindow));
end;
procedure TestRefreshWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 25', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    ClearWindow(testWindow, ColorWhite());
    DrawCircle(ColorBlack(), 400.0, 300.0, 50.0);
    RefreshWindow(testWindow);
    pixel := GetPixel(PointAt(400.0, 350.0));
    AssertEquals(ColorBlack(), pixel);
end;
procedure TestRefreshWindowWithTargetFpsIntegration;
begin
    testWindow := OpenWindow('Test Window 26', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    RefreshWindow(testWindow, Cardinal(60));
end;
procedure TestResizeCurrentWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 27', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    SetCurrentWindow(testWindow);
    ResizeCurrentWindow(1024, 768);
    AssertEquals(1024, WindowWidth(testWindow));
    AssertEquals(768, WindowHeight(testWindow));
end;
procedure TestResizeWindowIntegration;
begin
    testWindow := OpenWindow('Test Window 28', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    ResizeWindow(testWindow, 1024, 768);
    AssertEquals(1024, WindowWidth(testWindow));
    AssertEquals(768, WindowHeight(testWindow));
end;
procedure TestSetCurrentWindowNamedIntegration;
begin
    OpenWindow('Test Window 29', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    testWindow2 := OpenWindow('Test Window 30', 800, 600);
    SetCurrentWindow('Test Window 30');
    AssertEquals(testWindow2, CurrentWindow());
end;
procedure TestSetCurrentWindowIntegration;
begin
    OpenWindow('Test Window 31', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    testWindow2 := OpenWindow('Test Window 32', 800, 600);
    SetCurrentWindow(testWindow2);
    AssertEquals(testWindow2, CurrentWindow());
end;
procedure TestWindowCaptionIntegration;
begin
    testWindow := OpenWindow('Test Window 33', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertEquals('Test Window 33', WindowCaption(testWindow));
end;
procedure TestWindowCloseRequestedNamedIntegration;
begin
    testWindow := OpenWindow('Test Window 34', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertFalse(WindowCloseRequested('Test Window 34'));
    while WindowCloseRequested('Test Window 34') = false do
        ProcessEvents();
        ClearWindow(testWindow, ColorWhite());
        DrawText('Test: window_close_requested_named. Close window to pass.', ColorBlack(), 10.0, 10.0);
        RefreshScreen();
    end;
    AssertTrue(WindowCloseRequested('Test Window 34'));
end;
procedure TestWindowCloseRequestedIntegration;
begin
    testWindow := OpenWindow('Test Window 35', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertFalse(WindowCloseRequested(testWindow));
    while WindowCloseRequested(testWindow) = false do
        ProcessEvents();
        ClearWindow(testWindow, ColorWhite());
        DrawText('Test: window_close_requested. Close window to pass.', ColorBlack(), 10.0, 10.0);
        RefreshScreen();
    end;
    AssertTrue(WindowCloseRequested(testWindow));
end;
procedure TestWindowHasBorderNamedIntegration;
begin
    OpenWindow('Test Window 36', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertTrue(WindowHasBorder('Test Window 36'));
    WindowToggleBorder('Test Window 36');
    AssertFalse(WindowHasBorder('Test Window 36'));
end;
procedure TestWindowHasBorderIntegration;
begin
    testWindow := OpenWindow('Test Window 37', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertTrue(WindowHasBorder(testWindow));
    WindowToggleBorder(testWindow);
    AssertFalse(WindowHasBorder(testWindow));
end;
procedure TestWindowHasFocusIntegration;
begin
    testWindow1 := OpenWindow('Test Window 38', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    testWindow2 := OpenWindow('Test Window 39', 800, 600);
    ProcessEvents();
    AssertTrue(WindowHasFocus(testWindow2));
    AssertFalse(WindowHasFocus(testWindow1));
end;
procedure TestWindowHeightNamedIntegration;
begin
    testWindow := OpenWindow('Test Window 40', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertEquals(600, WindowHeight('Test Window 40'));
    ResizeWindow(testWindow, 800, 400);
    AssertEquals(400, WindowHeight('Test Window 40'));
end;
procedure TestWindowHeightIntegration;
begin
    testWindow := OpenWindow('Test Window 41', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertEquals(600, WindowHeight(testWindow));
    ResizeWindow(testWindow, 800, 400);
    AssertEquals(400, WindowHeight(testWindow));
end;
procedure TestWindowIsFullscreenNamedIntegration;
begin
    OpenWindow('Test Window 42', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertFalse(WindowIsFullscreen('Test Window 42'));
    WindowToggleFullscreen('Test Window 42');
    AssertTrue(WindowIsFullscreen('Test Window 42'));
end;
procedure TestWindowIsFullscreenIntegration;
begin
    testWindow := OpenWindow('Test Window 43', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertFalse(WindowIsFullscreen(testWindow));
    WindowToggleFullscreen(testWindow);
    AssertTrue(WindowIsFullscreen(testWindow));
end;
procedure TestWindowNamedIntegration;
begin
    testWindow := OpenWindow('Test Window 44', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    namedWindow := WindowNamed('Test Window 44');
    AssertEquals(testWindow, namedWindow);
    AssertEquals(WindowCaption(testWindow), WindowCaption(namedWindow));
end;
procedure TestWindowPositionNamedIntegration;
begin
    OpenWindow('Test Window 45', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo('Test Window 45', 100, 200);
    Delay(500);
    position := WindowPosition('Test Window 45');
    AssertEquals(68.0, position.x);
    AssertEquals(168.0, position.y);
end;
procedure TestWindowPositionIntegration;
begin
    testWindow := OpenWindow('Test Window 46', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo(testWindow, 100, 200);
    Delay(500);
    position := WindowPosition(testWindow);
    AssertEquals(68.0, position.x);
    AssertEquals(168.0, position.y);
end;
procedure TestWindowSetIconIntegration;
begin
    testWindow := OpenWindow('Test Window 47', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    icon := CreateBitmap('Test Bitmap 1', 32, 32);
    CleanupBitmap := TBitmapCleanup.Create;
    ClearBitmap(icon, ColorWhite());
    WindowSetIcon(testWindow, icon);
    FreeBitmap(icon);
end;
procedure TestWindowToggleBorderNamedIntegration;
begin
    OpenWindow('Test Window 48', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertTrue(WindowHasBorder('Test Window 48'));
    WindowToggleBorder('Test Window 48');
    AssertFalse(WindowHasBorder('Test Window 48'));
end;
procedure TestWindowToggleBorderIntegration;
begin
    testWindow := OpenWindow('Test Window 49', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertTrue(WindowHasBorder(testWindow));
    WindowToggleBorder(testWindow);
    AssertFalse(WindowHasBorder(testWindow));
end;
procedure TestWindowToggleFullscreenNamedIntegration;
begin
    OpenWindow('Test Window 50', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertFalse(WindowIsFullscreen('Test Window 50'));
    WindowToggleFullscreen('Test Window 50');
    AssertTrue(WindowIsFullscreen('Test Window 50'));
end;
procedure TestWindowToggleFullscreenIntegration;
begin
    testWindow := OpenWindow('Test Window 51', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertFalse(WindowIsFullscreen(testWindow));
    WindowToggleFullscreen(testWindow);
    AssertTrue(WindowIsFullscreen(testWindow));
end;
procedure TestWindowWidthNamedIntegration;
begin
    testWindow := OpenWindow('Test Window 52', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertEquals(800, WindowWidth('Test Window 52'));
    ResizeWindow(testWindow, 1024, 600);
    AssertEquals(1024, WindowWidth('Test Window 52'));
end;
procedure TestWindowWidthIntegration;
begin
    testWindow := OpenWindow('Test Window 53', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    AssertEquals(800, WindowWidth(testWindow));
    ResizeWindow(testWindow, 1024, 600);
    AssertEquals(1024, WindowWidth(testWindow));
end;
procedure TestWindowWithFocusIntegration;
begin
    testWindow1 := OpenWindow('Test Window 54', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo(testWindow1, 200, 200);
    Delay(500);
    testWindow2 := OpenWindow('Test Window 55', 800, 600);
    MoveWindowTo(testWindow2, 850, 200);
    Delay(500);
    AssertEquals(testWindow2, WindowWithFocus());
    while WindowWithFocus() <> testWindow1 do
        ProcessEvents();
        ClearScreen();
        DrawText('Test: window_with_focus. Click this window to pass.', ColorBlack(), 10.0, 10.0, OptionDrawTo(testWindow1));
        RefreshScreen();
    end;
    AssertEquals(testWindow1, WindowWithFocus());
end;
procedure TestWindowXNamedIntegration;
begin
    OpenWindow('Test Window 56', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo('Test Window 56', 100, 200);
    AssertEquals(68, WindowX('Test Window 56'));
    MoveWindowTo('Test Window 56', 300, 200);
    AssertEquals(268, WindowX('Test Window 56'));
end;
procedure TestWindowXIntegration;
begin
    testWindow := OpenWindow('Test Window 57', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo(testWindow, 100, 200);
    AssertEquals(68, WindowX(testWindow));
    MoveWindowTo(testWindow, 300, 200);
    AssertEquals(268, WindowX(testWindow));
end;
procedure TestWindowYNamedIntegration;
begin
    OpenWindow('Test Window 58', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo('Test Window 58', 100, 200);
    AssertEquals(168, WindowY('Test Window 58'));
    MoveWindowTo('Test Window 58', 100, 400);
    AssertEquals(368, WindowY('Test Window 58'));
end;
procedure TestWindowYIntegration;
begin
    testWindow := OpenWindow('Test Window 59', 800, 600);
    CleanupWindow := TWindowCleanup.Create;
    MoveWindowTo(testWindow, 100, 200);
    AssertEquals(168, WindowY(testWindow));
    MoveWindowTo(testWindow, 100, 400);
    AssertEquals(368, WindowY(testWindow));
end;
end;

procedure RegisterTests;
begin
    #<Proc:0x00007faa116e2450 /mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/test_generator/config/languages/pascal_config.rb:138 (lambda)>
end;

File: generated_tests/python/tests/windows_tests.py
import pytest
from splashkit import *
from ..helpers import *
import contextlib
class TestWindows:
    def test_clear_window_integration():
        test_window = open_window("Test Window 1", 800, 600)
        with window_cleanup():
            color = color_black()
            clear_window(test_window, color)
            refresh_window(test_window)
            pixel = get_pixel_at_point(point_at(0.0, 0.0))
            assert color == pixel
            
    def test_close_all_windows_integration():
        open_window("Test Window 2", 800, 600)
        with window_cleanup():
            open_window("Test Window 3", 800, 600)
            close_all_windows()
            assert not has_window("Test Window 3")
            assert not has_window("Test Window 3")
            
    def test_close_current_window_integration():
        test_window = open_window("Test Window 4", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            close_current_window()
            assert not has_window("Test Window 4")
            assert current_window() is None
            
    def test_close_window_named_integration():
        open_window("Test Window 5", 800, 600)
        with window_cleanup():
            assert has_window("Test Window 5")
            close_window_named("Test Window 5")
            assert not has_window("Test Window 5")
            
    def test_close_window_integration():
        test_window = open_window("Test Window 6", 800, 600)
        with window_cleanup():
            window_name = window_caption(test_window)
            close_window(test_window)
            assert not has_window(window_name)
            
    def test_current_window_integration():
        test_window1 = open_window("Test Window 7", 800, 600)
        with window_cleanup():
            test_window2 = open_window("Test Window 8", 800, 600)
            set_current_window(test_window2)
            assert test_window2 == current_window()
            assert is_current_window(test_window2)
            assert not is_current_window(test_window1)
            
    def test_current_window_has_border_integration():
        test_window = open_window("Test Window 9", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            assert current_window_has_border()
            current_window_toggle_border()
            assert not current_window_has_border()
            
    def test_current_window_height_integration():
        test_window = open_window("Test Window 10", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            assert 600 == current_window_height()
            resize_current_window(800, 400)
            assert 400 == current_window_height()
            
    def test_current_window_is_fullscreen_integration():
        test_window = open_window("Test Window 11", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            assert not current_window_is_fullscreen()
            current_window_toggle_fullscreen()
            assert current_window_is_fullscreen()
            
    def test_current_window_position_integration():
        test_window = open_window("Test Window 12", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            move_current_window_to(100, 200)
            position = current_window_position()
            assert 68.0 == position.x
            assert 168.0 == position.y
            
    def test_current_window_toggle_border_integration():
        test_window = open_window("Test Window 13", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            initial_border = current_window_has_border()
            current_window_toggle_border()
            assert initial_border != current_window_has_border()
            
    def test_current_window_toggle_fullscreen_integration():
        test_window = open_window("Test Window 14", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            initial_fullscreen = current_window_is_fullscreen()
            current_window_toggle_fullscreen()
            assert initial_fullscreen != current_window_is_fullscreen()
            
    def test_current_window_width_integration():
        test_window = open_window("Test Window 15", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            assert 800 == current_window_width()
            resize_current_window(400, 600)
            assert 400 == current_window_width()
            
    def test_current_window_x_integration():
        test_window = open_window("Test Window 16", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            move_current_window_to(100, 200)
            assert 68 == current_window_x()
            move_current_window_to(300, 200)
            assert 268 == current_window_x()
            
    def test_current_window_y_integration():
        test_window = open_window("Test Window 17", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            move_current_window_to(0, 100)
            assert 68 == current_window_y()
            
    def test_has_window_integration():
        test_window = open_window("Test Window 18", 800, 600)
        with window_cleanup():
            assert has_window("Test Window 18")
            assert has_window(window_caption(test_window))
            close_window(test_window)
            assert not has_window("Test Window 18")
            
    def test_is_current_window_integration():
        test_window1 = open_window("Test Window 19", 800, 600)
        with window_cleanup():
            test_window2 = open_window("Test Window 20", 800, 600)
            set_current_window(test_window1)
            assert is_current_window(test_window1)
            assert not is_current_window(test_window2)
            
    def test_move_current_window_to_integration():
        test_window = open_window("Test Window 21", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            move_current_window_to(100, 200)
            assert 68 == current_window_x()
            assert 168 == current_window_y()
            
    def test_move_window_to_named_integration():
        open_window("Test Window 22", 800, 600)
        with window_cleanup():
            move_window_to_named("Test Window 22", 150, 250)
            assert 118 == window_x_named("Test Window 22")
            assert 218 == window_y_named("Test Window 22")
            
    def test_move_window_to_integration():
        test_window = open_window("Test Window 23", 800, 600)
        with window_cleanup():
            move_window_to(test_window, 200, 300)
            assert 168 == window_x(test_window)
            assert 268 == window_y(test_window)
            
    def test_open_window_integration():
        test_window = open_window("Test Window 24", 800, 600)
        with window_cleanup():
            assert test_window is not None
            assert "Test Window 24" == window_caption(test_window)
            assert 800 == window_width(test_window)
            assert 600 == window_height(test_window)
            
    def test_refresh_window_integration():
        test_window = open_window("Test Window 25", 800, 600)
        with window_cleanup():
            clear_window(test_window, color_white())
            draw_circle(color_black(), 400.0, 300.0, 50.0)
            refresh_window(test_window)
            pixel = get_pixel_at_point(point_at(400.0, 350.0))
            assert color_black() == pixel
            
    def test_refresh_window_with_target_fps_integration():
        test_window = open_window("Test Window 26", 800, 600)
        with window_cleanup():
            refresh_window_with_target_fps(test_window, 60)
            
    def test_resize_current_window_integration():
        test_window = open_window("Test Window 27", 800, 600)
        with window_cleanup():
            set_current_window(test_window)
            resize_current_window(1024, 768)
            assert 1024 == window_width(test_window)
            assert 768 == window_height(test_window)
            
    def test_resize_window_integration():
        test_window = open_window("Test Window 28", 800, 600)
        with window_cleanup():
            resize_window(test_window, 1024, 768)
            assert 1024 == window_width(test_window)
            assert 768 == window_height(test_window)
            
    def test_set_current_window_named_integration():
        open_window("Test Window 29", 800, 600)
        with window_cleanup():
            test_window2 = open_window("Test Window 30", 800, 600)
            set_current_window_named("Test Window 30")
            assert test_window2 == current_window()
            
    def test_set_current_window_integration():
        open_window("Test Window 31", 800, 600)
        with window_cleanup():
            test_window2 = open_window("Test Window 32", 800, 600)
            set_current_window(test_window2)
            assert test_window2 == current_window()
            
    def test_window_caption_integration():
        test_window = open_window("Test Window 33", 800, 600)
        with window_cleanup():
            assert "Test Window 33" == window_caption(test_window)
            
    def test_window_close_requested_named_integration():
        test_window = open_window("Test Window 34", 800, 600)
        with window_cleanup():
            assert not window_close_requested_named("Test Window 34")
        while window_close_requested_named("Test Window 34") is False:
            process_events()
            clear_window(test_window, color_white())
            draw_text_no_font_no_size("Test: window_close_requested_named. Close window to pass.", color_black(), 10.0, 10.0)
            refresh_screen()
            
            assert window_close_requested_named("Test Window 34")
            
    def test_window_close_requested_integration():
        test_window = open_window("Test Window 35", 800, 600)
        with window_cleanup():
            assert not window_close_requested(test_window)
        while window_close_requested(test_window) is False:
            process_events()
            clear_window(test_window, color_white())
            draw_text_no_font_no_size("Test: window_close_requested. Close window to pass.", color_black(), 10.0, 10.0)
            refresh_screen()
            
            assert window_close_requested(test_window)
            
    def test_window_has_border_named_integration():
        open_window("Test Window 36", 800, 600)
        with window_cleanup():
            assert window_has_border_named("Test Window 36")
            window_toggle_border_named("Test Window 36")
            assert not window_has_border_named("Test Window 36")
            
    def test_window_has_border_integration():
        test_window = open_window("Test Window 37", 800, 600)
        with window_cleanup():
            assert window_has_border(test_window)
            window_toggle_border(test_window)
            assert not window_has_border(test_window)
            
    def test_window_has_focus_integration():
        test_window1 = open_window("Test Window 38", 800, 600)
        with window_cleanup():
            test_window2 = open_window("Test Window 39", 800, 600)
            process_events()
            assert window_has_focus(test_window2)
            assert not window_has_focus(test_window1)
            
    def test_window_height_named_integration():
        test_window = open_window("Test Window 40", 800, 600)
        with window_cleanup():
            assert 600 == window_height_named("Test Window 40")
            resize_window(test_window, 800, 400)
            assert 400 == window_height_named("Test Window 40")
            
    def test_window_height_integration():
        test_window = open_window("Test Window 41", 800, 600)
        with window_cleanup():
            assert 600 == window_height(test_window)
            resize_window(test_window, 800, 400)
            assert 400 == window_height(test_window)
            
    def test_window_is_fullscreen_named_integration():
        open_window("Test Window 42", 800, 600)
        with window_cleanup():
            assert not window_is_fullscreen_named("Test Window 42")
            window_toggle_fullscreen_named("Test Window 42")
            assert window_is_fullscreen_named("Test Window 42")
            
    def test_window_is_fullscreen_integration():
        test_window = open_window("Test Window 43", 800, 600)
        with window_cleanup():
            assert not window_is_fullscreen(test_window)
            window_toggle_fullscreen(test_window)
            assert window_is_fullscreen(test_window)
            
    def test_window_named_integration():
        test_window = open_window("Test Window 44", 800, 600)
        with window_cleanup():
            named_window = window_named("Test Window 44")
            assert test_window == named_window
            assert window_caption(test_window) == window_caption(named_window)
            
    def test_window_position_named_integration():
        open_window("Test Window 45", 800, 600)
        with window_cleanup():
            move_window_to_named("Test Window 45", 100, 200)
            delay(500)
            position = window_position_named("Test Window 45")
            assert 68.0 == position.x
            assert 168.0 == position.y
            
    def test_window_position_integration():
        test_window = open_window("Test Window 46", 800, 600)
        with window_cleanup():
            move_window_to(test_window, 100, 200)
            delay(500)
            position = window_position(test_window)
            assert 68.0 == position.x
            assert 168.0 == position.y
            
    def test_window_set_icon_integration():
        test_window = open_window("Test Window 47", 800, 600)
        with window_cleanup():
            icon = create_bitmap("Test Bitmap 1", 32, 32)
            with bitmap_cleanup():
                clear_bitmap(icon, color_white())
                window_set_icon(test_window, icon)
                free_bitmap(icon)
                
    def test_window_toggle_border_named_integration():
        open_window("Test Window 48", 800, 600)
        with window_cleanup():
            assert window_has_border_named("Test Window 48")
            window_toggle_border_named("Test Window 48")
            assert not window_has_border_named("Test Window 48")
            
    def test_window_toggle_border_integration():
        test_window = open_window("Test Window 49", 800, 600)
        with window_cleanup():
            assert window_has_border(test_window)
            window_toggle_border(test_window)
            assert not window_has_border(test_window)
            
    def test_window_toggle_fullscreen_named_integration():
        open_window("Test Window 50", 800, 600)
        with window_cleanup():
            assert not window_is_fullscreen_named("Test Window 50")
            window_toggle_fullscreen_named("Test Window 50")
            assert window_is_fullscreen_named("Test Window 50")
            
    def test_window_toggle_fullscreen_integration():
        test_window = open_window("Test Window 51", 800, 600)
        with window_cleanup():
            assert not window_is_fullscreen(test_window)
            window_toggle_fullscreen(test_window)
            assert window_is_fullscreen(test_window)
            
    def test_window_width_named_integration():
        test_window = open_window("Test Window 52", 800, 600)
        with window_cleanup():
            assert 800 == window_width_named("Test Window 52")
            resize_window(test_window, 1024, 600)
            assert 1024 == window_width_named("Test Window 52")
            
    def test_window_width_integration():
        test_window = open_window("Test Window 53", 800, 600)
        with window_cleanup():
            assert 800 == window_width(test_window)
            resize_window(test_window, 1024, 600)
            assert 1024 == window_width(test_window)
            
    def test_window_with_focus_integration():
        test_window1 = open_window("Test Window 54", 800, 600)
        with window_cleanup():
            move_window_to(test_window1, 200, 200)
            delay(500)
            test_window2 = open_window("Test Window 55", 800, 600)
            move_window_to(test_window2, 850, 200)
            delay(500)
            assert test_window2 == window_with_focus()
        while window_with_focus() != test_window1:
            process_events()
            clear_screen_to_white()
            draw_text_no_font_no_size_with_options("Test: window_with_focus. Click this window to pass.", color_black(), 10.0, 10.0, option_draw_to_window(test_window1))
            refresh_screen()
            
            assert test_window1 == window_with_focus()
            
    def test_window_x_named_integration():
        open_window("Test Window 56", 800, 600)
        with window_cleanup():
            move_window_to_named("Test Window 56", 100, 200)
            assert 68 == window_x_named("Test Window 56")
            move_window_to_named("Test Window 56", 300, 200)
            assert 268 == window_x_named("Test Window 56")
            
    def test_window_x_integration():
        test_window = open_window("Test Window 57", 800, 600)
        with window_cleanup():
            move_window_to(test_window, 100, 200)
            assert 68 == window_x(test_window)
            move_window_to(test_window, 300, 200)
            assert 268 == window_x(test_window)
            
    def test_window_y_named_integration():
        open_window("Test Window 58", 800, 600)
        with window_cleanup():
            move_window_to_named("Test Window 58", 100, 200)
            assert 168 == window_y_named("Test Window 58")
            move_window_to_named("Test Window 58", 100, 400)
            assert 368 == window_y_named("Test Window 58")
            
    def test_window_y_integration():
        test_window = open_window("Test Window 59", 800, 600)
        with window_cleanup():
            move_window_to(test_window, 100, 200)
            assert 168 == window_y(test_window)
            move_window_to(test_window, 100, 400)
            assert 368 == window_y(test_window)
            
